### PR TITLE
feat: Generate nicer-looking SQL when grouping by expressions

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Like this we get
```sql
WITH "t27701520" AS (
  SELECT
    "penguins"."island_name" AS "_island_name_tmp",
    ROUND("penguins"."height", 10) AS "_height_tmp_",
    AVG("penguins"."weight") AS "weight"
  FROM "penguins" AS "penguins"
  JOIN "islands" AS "islands"
    ON "islands"."name" = "penguins"."island_name"
  GROUP BY
    "penguins"."island_name",
    ROUND("penguins"."height", 10)
)
SELECT
  "t27701520"."_island_name_tmp" AS "island_name",
  "t27701520"."_height_tmp_" AS "height",
  "t27701520"."weight" AS "weight"
FROM "t27701520" AS "t27701520"
```

instead of
```sql
WITH "t36714377" AS (
  SELECT
    "penguins"."island_name" AS "island_name____________",
    ROUND("penguins"."height", 10) AS "height____________",
    AVG("penguins"."weight") AS "weight"
  FROM "penguins" AS "penguins"
  JOIN "islands" AS "islands"
    ON "islands"."name" = "penguins"."island_name"
  GROUP BY
    "penguins"."island_name",
    ROUND("penguins"."height", 10)
)
SELECT
  "t36714377"."island_name____________" AS "island_name",
  "t36714377"."height____________" AS "height",
  "t36714377"."weight" AS "weight"
FROM "t36714377" AS "t36714377"
```

## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📝 Documentation
- [ ] ✅ Test
- [ ] 🐳 Other

## Related issues

- Related issue #\<issue number\>
- Closes #\<issue number\>

## Checklist

- [ ] Code follows style guide (ruff)
- [ ] Tests added
- [ ] Documented the changes

### CODE REVIEW

File: tests/group_by_test.py [hunk 1]
```diff
@@ -584,3 +584,10 @@ def test_group_by_selector(constructor: Constructor) -> None:
     )
     expected = {"a": [1, 1], "b": [4, 6], "c": [8.0, 9.0]}
     assert_equal_data(result, expected)
+
+
+def test_renaming_edge_case(constructor: Constructor) -> None:
+    data = {"a": [0, 0, 0], "_a_tmp": [1, 2, 3], "b": [4, 5, 6]}
+    result = nw.from_native(constructor(data)).group_by(nw.col("a")).agg(nw.all().min())
+    expected = {"a": [0], "_a_tmp": [1], "b": [4]}
+    assert_equal_data(result, expected)
```
Comment (FBruzzesi) [inline tests/group_by_test.py:593]: It should be safe as it will have at least one additional character i.e. the new column generated is `_a_tmp_`

**Review by FBruzzesi — APPROVED** [review]
Thanks @MarcoGorelli 

SwiftmoTwitchGIF

File: tests/group_by_test.py (hunk 1 above)
Comment (MarcoGorelli) [reply tests/group_by_test.py:593]: 😄 I wanted a test so that if anyone tried removing the `+1` then this would stop them

Comment (FBruzzesi) [reply tests/group_by_test.py:593]: Yep 100% worth it to have edge case tests
